src/main/kotlin/org/rust/ide/annotator/RsMutableSelfParameterAnnotator.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.psi.PsiElement
+import org.rust.ide.colors.RsColor
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.RsSelfParameter
+import org.rust.lang.core.psi.ext.mutability
+import org.rust.lang.core.types.ty.Mutability
+
+class RsMutableSelfParameterAnnotator : Annotator {
+    private val LOW_PRIORITY_HIGHLIGHTING = HighlightSeverity(
+        "LOW_PRIORITY_HIGHLIGHTING",
+        HighlightSeverity.INFORMATION.myVal - 3
+    )
+
+    override fun annotate(element: PsiElement, holder: AnnotationHolder) {
+        if (element is RsPath) {
+            if (element.self != null) {
+                val ref = element.reference.resolve() as? RsSelfParameter ?: return
+                if (ref.mutability == Mutability.MUTABLE) {
+                    addHighlightingAnnotation(holder, element, RsColor.MUT_PARAMETER, LOW_PRIORITY_HIGHLIGHTING)
+                }
+            }
+        }
+        if (element is RsSelfParameter) {
+            if (element.mutability == Mutability.MUTABLE) {
+                addHighlightingAnnotation(holder, element.self, RsColor.MUT_PARAMETER, LOW_PRIORITY_HIGHLIGHTING)
+            }
+        }
+    }
+
+    private fun addHighlightingAnnotation(holder: AnnotationHolder,
+                                          target: PsiElement,
+                                          key: RsColor,
+                                          severity: HighlightSeverity = HighlightSeverity.INFORMATION) {
+        val testMode = ApplicationManager.getApplication().isUnitTestMode
+        val actualSeverity = if (testMode && severity.myVal < HighlightSeverity.INFORMATION.myVal) {
```

Comment:
Let's add a comment why we change annotation priority here because it isn't obvious